fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/ContentObject.java

```diff
@@ -31,11 +37,55 @@ protected ContentObject(PID pid, Repository repository, RepositoryObjectDataLoad
 		super(pid, repository, dataLoader);
 	}
 
-	public BinaryObject addDescription(InputStream contentStream) {
-		return null;
+	public FileObject addDescription(InputStream modsStream) {
+		FileObject fileObj = createFileObject();
+		
+		BinaryObject mods = fileObj.addOriginalFile(modsStream, null, "text/xml", null);
+		repository.createRelationship(pid, PcdmModels.hasRelatedObject, fileObj.getResource());
+		repository.createRelationship(pid, Cdr.hasMods, mods.getResource());
+		return fileObj;
+	}
+	
+	public FileObject addDescription(InputStream sourceMdStream, String sourceProfile,
+			InputStream modsStream) {
+		FileObject fileObj = createFileObject();
+		
+		BinaryObject orig = fileObj.addOriginalFile(sourceMdStream, null, "text/plain", null);
+		repository.createRelationship(orig.getPid(), RDF.type, Cdr.SourceMetadata);
+		repository.createRelationship(pid, PcdmModels.hasRelatedObject, fileObj.getResource());
+		orig.getResource().addProperty(Cdr.hasSourceMetadataProfile, sourceProfile);
+		
+		BinaryObject mods = fileObj.addDerivative(null, modsStream, null, "text/plain", null);
+		repository.createRelationship(pid, Cdr.hasMods, mods.getResource());
+		
+		return fileObj;
 	}
 
-	public BinaryObject getDescription() {
-		return null;
+	public FileObject getDescription() {
+		Resource sourceMd = this.getResource();
```

Comment:
I'd suggest renaming this variable since sourceMd implies source metadata in this context, but its actually just the resource for the current contentObject.